### ISSUE TITLE
MetadataFileReference is internal

### ISSUE THREAD

**Issue by provegard** [dotnet/roslyn#932]
I'm trying to follow `csharp-semantic.pdf` under docs/samples. When I add this code:

```
var compilation = CSharpCompilation.Create("HelloWorld")
 .AddReferences(
   new MetadataFileReference(typeof(object).Assembly.Location))
 .AddSyntaxTrees(tree);
```

I get a compilation error because MetadataFileReference is an internal type.

**Comment by provegard** [comment]
Got it to work using `MetadataReference.CreateFromAssembly(typeof (object).Assembly)`, so I suppose the sample doc should be updated.

**Comment by Pilchie** [comment]
Yes, this is deliberately exposed through the factory methods now.  There are (unfortunately) a lot of things that need to be updated in that document.

**Comment by gafter** [comment]
Reopening as the docs are still wrong.

**Comment by CyrusNajmabadi** [comment]
Moving to docs team to decide future of doc'ing roslyn.
